projects/aws-observability/aws-otel-collector/helm/patches/0002-namespace-patch.yaml

```diff
@@ -1,193 +1,222 @@
-Date: Wed, 24 Aug 2022 07:08:19 -0500
+Date: Wed, 14 Sep 2022 10:44:03 -0500
 Subject: [PATCH] Add namespace to charts
 
 ---
- charts/opentelemetry-collector/templates/configmap-agent.yaml    | 1 +
- .../opentelemetry-collector/templates/configmap-statefulset.yaml | 1 +
- charts/opentelemetry-collector/templates/configmap.yaml          | 1 +
- charts/opentelemetry-collector/templates/daemonset.yaml          | 1 +
- charts/opentelemetry-collector/templates/deployment.yaml         | 1 +
- charts/opentelemetry-collector/templates/hpa.yaml                | 1 +
- charts/opentelemetry-collector/templates/ingress.yaml            | 1 +
- charts/opentelemetry-collector/templates/pdb.yaml                | 1 +
- charts/opentelemetry-collector/templates/podmonitor.yaml         | 1 +
- charts/opentelemetry-collector/templates/prometheusrule.yaml     | 1 +
- charts/opentelemetry-collector/templates/service.yaml            | 1 +
- charts/opentelemetry-collector/templates/serviceaccount.yaml     | 1 +
- charts/opentelemetry-collector/templates/servicemonitor.yaml     | 1 +
- charts/opentelemetry-collector/templates/statefulset.yaml        | 1 +
- 14 files changed, 14 insertions(+)
+ charts/opentelemetry-collector/templates/configmap-agent.yaml | 1 +
+ .../templates/configmap-statefulset.yaml                      | 1 +
+ charts/opentelemetry-collector/templates/configmap.yaml       | 1 +
+ charts/opentelemetry-collector/templates/daemonset.yaml       | 1 +
+ charts/opentelemetry-collector/templates/deployment.yaml      | 1 +
+ charts/opentelemetry-collector/templates/hpa.yaml             | 1 +
+ charts/opentelemetry-collector/templates/ingress.yaml         | 1 +
+ charts/opentelemetry-collector/templates/pdb.yaml             | 1 +
+ charts/opentelemetry-collector/templates/podmonitor.yaml      | 1 +
+ charts/opentelemetry-collector/templates/prometheusrule.yaml  | 1 +
+ charts/opentelemetry-collector/templates/service.yaml         | 1 +
+ charts/opentelemetry-collector/templates/serviceaccount.yaml  | 1 +
+ charts/opentelemetry-collector/templates/servicemonitor.yaml  | 1 +
+ charts/opentelemetry-collector/templates/statefulset.yaml     | 1 +
+ charts/opentelemetry-collector/values.schema.json             | 4 ++++
+ charts/opentelemetry-collector/values.yaml                    | 1 +
+ 16 files changed, 19 insertions(+)
 
 diff --git a/charts/opentelemetry-collector/templates/configmap-agent.yaml b/charts/opentelemetry-collector/templates/configmap-agent.yaml
-index e267336..d673a40 100644
+index e267336..2658f7b 100644
 --- a/charts/opentelemetry-collector/templates/configmap-agent.yaml
 +++ b/charts/opentelemetry-collector/templates/configmap-agent.yaml
 @@ -3,6 +3,7 @@ apiVersion: v1
  kind: ConfigMap
  metadata:
    name: {{ include "opentelemetry-collector.fullname" . }}-agent
-+  namespace: {{ .Release.Namespace | default "observability" | quote }}
++  namespace: {{ .Release.Namespace | default .Values.defaultNamespace | quote }}
```

Comment:
Do we have a standard across charts for this. I think in the CRD, we use `targetNamespace` and others probably use `namespace`? The name `defaultNamespace` seems a bit odd because it looks like it is actually the namespace.

Reply:
@junshun Steven will have more context on this. It is a standard across our existing packages today. See [metallb](https://github.com/aws/eks-anywhere-build-tooling/blob/main/projects/metallb/metallb/helm/patches/0002-Adding-imagePullSecrets-and-extracting-default-names.patch) as an example.

Reply:
Currently the defaultNamespace value is used here so the controller knows which namespace the chart will install to in the cases where the user does not provide a targetNamespace. This info is used so we can add the ecr-token to that namespace.

Reply:
Okay, defaultNamespace makes sense. We need to make sure we use that across the other projects.